### ISSUE TITLE
TFC: Reflect nesting levels (module / resource type / instance)

### ISSUE THREAD

**Issue by radeksimko** [hashicorp/vscode-terraform#1638]
## Background

In https://github.com/hashicorp/vscode-terraform/pull/1590 we introduced a panel which displays structured plan log in the form of a tree. For example:

![Screenshot 2023-11-20 at 15 00 32](https://github.com/hashicorp/vscode-terraform/assets/287584/cde42e50-7d54-4062-879e-f9c60a150ccb)

As shown in the screenshot ^, such a tree can be difficult to navigate if there are many resources in the plan.

## Proposal

 - [ ] Consider folding resources under their respective module (e.g. anything like `module.workspaces.*` under `module.workspaces` per above example)
 - [ ] Consider folding resource instances under their respective resource types (e.g. anything like `*.tfe_workspace.config_a[*]` under `tfe_workspace.config_a`)

## Implementation Notes

The big hurdle and main reason this was omitted from https://github.com/hashicorp/vscode-terraform/pull/1590 initially is because the nesting introduces a new problem of mixed actions. For example, one resource instance under the same module or type can be created, another modified and yet another deleted. It is unclear how to best represent such a scenario in the tree view and whether we should be showing a particular icon or avoid attaching any icons at all.

TL;DR The UX implications here is the hard part.
